### PR TITLE
refactor, add more tests

### PR DESCRIPTION
**Pull request by rz-comp**
- adapt structure of pattern recognition code to make it testable
- add more tests, what's missing now is mostly to have more test documents
- fix/improve some doc strings

### CODE REVIEW

**Comment by ecino** [comment]
:+1:

File: sbc_compassion/tools/bluecornerfinder.py [hunk 1]
```diff
@@ -119,42 +116,8 @@ def _readBlueCorner(self):
...
-
-        # case where a blue pixel has been found
-        if self.found:
-            self._findmin()
-        else:
-            self.ind = None
-
-        return self.ind
-
-    def _findmin(self):
```
Comment (loikki) [inline sbc_compassion/tools/bluecornerfinder.py]: I think we should keep this function in order to take the closest pixel and not one next to it

Comment (ecino) [inline sbc_compassion/tools/bluecornerfinder.py]: Please put back this function

File: sbc_compassion/tools/bluecornerfinder.py [hunk 2]
```diff
@@ -137,18 +132,18 @@ def _findmin(self):
...
-            for j in range(self.w - self.n - 1, self.w):
+            for j in range(max(0, self.w - self.n - 1), self.w):
                 # check if it is on the good ring
                 if self.dist[i, j] == self.n:
                     # check the color of the pixel
                     if checkColor(self.img[i, j], self.threshold):
                         # compute the exact distance (float not int)
                         tmp = np.sqrt(i ** 2 + (self.w - j - 1) ** 2)
-                        # check if closer and has at least 2 neighbours
+                        # check if closer and has at least 2 neighbors
```
Comment (loikki) [inline sbc_compassion/tools/bluecornerfinder.py]: Need to revert this change :P

**Comment by ecino** [comment]
Branch needs to be rebased to fix conflicts

**Comment by rz-comp** [comment]
rebase done

**Comment by ecino** [comment]
:+1:
